### PR TITLE
sphinx: drop python 2 + update jupyter.

### PR DESCRIPTION
**Pull request by sgn**
- Series to drop Python 2 support of Sphinx,
- as of it's now, only Jupyter requires python 2 version of Sphinx

### CODE REVIEW

File: srcpkgs/ipython/template [hunk 1]
```diff
@@ -14,21 +14,17 @@ maintainer="Alessio Sergi <al3hex@gmail.com>"
...
- ipython:ipython:/usr/bin/ipython3
- ipython:ipython.1:/usr/share/man/man1/ipython3.1"
+checksum=ca478e52ae1f88da0102360e57e528b92f3ae4316aabac80a2cd7f7ab2efb48a
+conflicts="python-ipython>=0"
+replaces="python3-ipython>=0"
 
 post_install() {
 	# use decorators of numpy
-	rm -f ${DESTDIR}/usr/lib/python*/site-packages/IPython/external/decorators/_decorators.py
+	: rm -f ${DESTDIR}/${py3_sitelib}/IPython/external/decorators/_decorators.py
```
[the remark this answers is not in the document]
Comment (sgn) [reply srcpkgs/ipython/template]: It was added during do_check, I forgot to remove it.

File: srcpkgs/ipython/template [hunk 2]
```diff
@@ -14,21 +14,17 @@ maintainer="Alessio Sergi <al3hex@gmail.com>"
 license="BSD-3-Clause"
 homepage="https://ipython.org/"
 distfiles="${PYPI_SITE}/i/ipython/ipython-${version}.tar.gz"
-checksum=d9459e7237e2e5858738ff9c3e26504b79899b58a6d49e574d352493d80684c6
-
-alternatives="
- ipython:ipython:/usr/bin/ipython3
- ipython:ipython.1:/usr/share/man/man1/ipython3.1"
+checksum=ca478e52ae1f88da0102360e57e528b92f3ae4316aabac80a2cd7f7ab2efb48a
+conflicts="python-ipython>=0"
```
[the remark this answers is not in the document]
Comment (sgn) [reply srcpkgs/ipython/template]: Both,

- `python-ipython` provides an alternate for `/usr/bin/ipython`,
- `ipython` provides an alternate for `/usr/bin/ipython`

It should be a conflict, and it doesn't make sense to make that a `python3` package can `replaces` a python2 one.

If you think it's better to be a replaces, we can do so.

File: srcpkgs/jupyter_notebook/template [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	: no selenium
+}
+
+jupyter_package() {
+	build_style=meta
+	archs=noarch
+	depends="jupyter_notebook jupyter_console python3-jupyter_ipywidgets"
+	short_desc="Jupyter metapackage for Void Linux"
+	conflicts="python-jupyter>=0"
+	replaces="python3-jupyter>=0"
```
[the remark this answers is not in the document]
Comment (sgn) [reply srcpkgs/jupyter_notebook/template]: Which action will be better? Add transitional package or keep the old name?

File: srcpkgs/ipython/template [hunk 4]
```diff
@@ -14,21 +14,17 @@ maintainer="Alessio Sergi <al3hex@gmail.com>"
 license="BSD-3-Clause"
 homepage="https://ipython.org/"
 distfiles="${PYPI_SITE}/i/ipython/ipython-${version}.tar.gz"
-checksum=d9459e7237e2e5858738ff9c3e26504b79899b58a6d49e574d352493d80684c6
-
-alternatives="
- ipython:ipython:/usr/bin/ipython3
```
[the remark this answers is not in the document]
Comment (sgn) [reply srcpkgs/ipython/template]: It seems to work in my case.

File: srcpkgs/jupyter_notebook/template (hunk 3 above)
[the remark this answers is not in the document]
Comment (Chocimier) [reply srcpkgs/jupyter_notebook/template]: Transitional package is fine.

**Comment by Chocimier** [comment]
I see PR is changed often, is it still in progress, or just updated versions and ready to review?

**Comment by sgn** [comment]
It's ready, but whenever I pushed it out, something was changed in Python core and Travis is failing.

I push again to try to address it.

**Comment by Chocimier** [comment]
Python is available now, please rebase.

**Comment by sgn** [comment]
Ok, Travis is passed.

File: srcpkgs/python3-jupyter_ipywidgets/template [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'python3-jupyter_ipywidgets'
+pkgname=python3-jupyter_ipywidgets
+version=7.5.1
+revision=1
+archs=noarch
+_widgetsnbextver=3.4.0
+wrksrc="ipywidgets-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-jupyter_widgetsnbextension>=${_widgetsnbextver}
```
[the remark this answers is not in the document]
Comment (sgn) [reply srcpkgs/python3-jupyter_ipywidgets/template]: Bumped together with version bump.

**Comment by Chocimier** [comment]
```
=> python3-jupyter_client-6.1.0_1: building [python3-module] (dependency of python3-ipython_ipykernel) for x86_64...
   [host] python3-setuptools-46.0.0_1: found (https://alpha.de.repo.voidlinux.org/current)
=> ERROR: python3-jupyter_client-6.1.0_1: check dependency 'python3-jupyter_core>=4.6.0' does not exist!
```

Checkdepends can't have version specified, bash is incapable of removing them, and using external command here is too costly, please repeat depends there. Xbps-src will take version present in template.

**Comment by sgn** [comment]
> ```
> => python3-jupyter_client-6.1.0_1: building [python3-module] (dependency of python3-ipython_ipykernel) for x86_64...
>    [host] python3-setuptools-46.0.0_1: found (https://alpha.de.repo.voidlinux.org/current)
> => ERROR: python3-jupyter_client-6.1.0_1: check dependency 'python3-jupyter_core>=4.6.0' does not exist!
> ```
> 
> Checkdepends can't have version specified,

Somehow, it worked in my local machine.
I dun know.

> bash is incapable of removing them, and using external command here is too costly, please repeat depends there. Xbps-src will take version present in template.

**Comment by sgn** [comment]
rebase --ignore-date to have better topology, github's topology is hard to follow.

**Comment by sgn** [comment]
Forced push to fix conflicts.

**Comment by Chocimier** [comment]
Tests are significant part and could be removed from python3-jupyter_notebook, python3-QtPy, python3-ipython_genutils, python3-jupyter_core.

jupyter_trust (in nbformat), jupyter_console executables are lost in updated version, not sure if removed deliberately, or is it upstream bug.
